call-home/src/transmitter/encrypt.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{common::constants::*, Report};
+use mktemp::Temp;
+use rand::{distributions::Alphanumeric, Rng};
+use serde_json::json;
+use std::{
+    fs,
+    io::{Error, ErrorKind},
+    process::Command,
+};
+use tracing::debug;
+
+/// 'encrypt' accepts a crate::collector::Report, marshals it into
+/// JSON and encrypts it.
+pub(crate) fn encrypt(report: &Report) -> Result<Vec<u8>, Error> {
+    // The underlying filesystem resource is garbage collected
+    // when this function returns.
+    // Ref: https://docs.rs/mktemp/0.4.1/mktemp/index.html
+    let temp_file = Temp::new_file_in(DATA_DIR_PATH)?;
+    debug!("Successfully created temporary input file.");
+
+    let path_buffer = temp_file.to_path_buf();
+    let input_filepath: &str = path_buffer
+        .to_str()
+        .ok_or_else(|| Error::new(ErrorKind::Other, "Error converting Pathbuf to &str"))?;
+    fs::write(
+        input_filepath,
+        <String as AsRef<[u8]>>::as_ref(&json!(report).to_string()),
+    )?;
+    debug!("Successfully written Report data to temporary input file.");
+
+    let random_name: String = rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(32)
+        .map(char::from)
+        .collect();
+    let output_filepath = DATA_DIR_PATH.to_owned() + "/" + &random_name + ".gpg";
+
+    let command = format!("gpg --yes --trust-model=always --homedir={} --keyring={} --recipient=product-health-report@caringo.com --no-default-keyring --encrypt -z=9 --output={} {}",
+                          DATA_DIR_PATH, ENCRYPTION_KEY_FILEPATH,
+                          &output_filepath,
+                          &input_filepath);
```

Comment:
do we need an external command for this? Is there no library for the effect?

Reply:
There is no official library. There were a couple of zlib compression libraries.
I'll add a TODO comment here. This is likely tech debt.

Reply:
The office gnupg-me library does not have a compression library.

Reply:
ok, should we pin a specific version of gpg or is that api stable enough that there won't be changes to cli arguments?

Reply:
We may chose to go with the latest one, that has been tested to work well with the receiver API's code.

Reply:
cc: @datacore-vkumar